Libraries/Opc.Ua.Security.Certificates/PEM/PEMReader.cs

```diff
@@ -0,0 +1,181 @@
+                }
+                else
+                {
+                    Password pwFinder = new Password(password.ToCharArray());
+                    pemReader = new Org.BouncyCastle.OpenSsl.PemReader(pemStreamReader, pwFinder);
+                }
+                try
+                {
+                    // find the private key in the PEM blob
+                    var pemObject = pemReader.ReadObject();
+                    while (pemObject != null)
+                    {
+                        RsaPrivateCrtKeyParameters privateKey = null;
+                        var keypair = pemObject as Org.BouncyCastle.Crypto.AsymmetricCipherKeyPair;
+                        if (keypair != null)
+                        {
+                            privateKey = keypair.Private as RsaPrivateCrtKeyParameters;
+                            break;
+                        }
+
+                        if (privateKey == null)
+                        {
+                            privateKey = pemObject as RsaPrivateCrtKeyParameters;
+                        }
+
+                        if (privateKey != null)
+                        {
+                            rsaPrivateKey = RSA.Create();
+                            rsaPrivateKey.ImportParameters(DotNetUtilities.ToRSAParameters(privateKey));
+                            break;
+                        }
+
+                        // read next object
+                        pemObject = pemReader.ReadObject();
+                    }
+                }
+                finally
+                {
+                    pemReader.Reader.Dispose();
+                }
+            }
+
+            if (rsaPrivateKey == null)
+            {
+                throw new CryptographicException("PEM data blob does not contain a private key.");
+            }
+
+            return rsaPrivateKey;
+        }
+#else
+        public static RSA ImportPrivateKeyFromPEM(
+            byte[] pemDataBlob,
+            string password = null)
+        {
+            string[] labels = {
+                "ENCRYPTED PRIVATE KEY", "PRIVATE KEY", "RSA PRIVATE KEY"
+                };
+            try
+            {
+                string pemText = Encoding.UTF8.GetString(pemDataBlob);
+                int count = 0;
+                foreach (var label in labels)
+                {
+                    count++;
+                    string beginlabel = $"-----BEGIN {label}-----";
+                    int beginIndex = pemText.IndexOf(beginlabel);
+                    if (beginIndex < 0)
+                    {
+                        continue;
+                    }
+                    string endlabel = $"-----END {label}-----";
+                    int endIndex = pemText.IndexOf(endlabel);
+                    beginIndex += beginlabel.Length;
+                    if (endIndex < 0 || endIndex <= beginIndex)
+                    {
+                        continue;
```

Comment:
Adding error log into here? It is not valid that we have _BEGIN {Label}_ without _END {Label}_